### ISSUE TITLE
Restocks Monitor Feedback

### ISSUE THREAD

**Issue by manbearcoon** [val92130/Supreme-Auto-Checkout#56]
Thanks for the new release. Few things that I noticed:

- I'm receiving multiple toast notifications for items which have restocked, typically 3-4 times. They show up every few minutes. Wondering if that's a bug.
- The restock toast notification doesn't show the size of the item which restocked, can that be added?

**Comment by val92130** [comment]
Hey, thanks for the feedback! You should only receive notifications for items that restocks, if you receive the notifications multiple times it's that it probably restocked multiple times on the Supreme store.

For the moment I don't plan on monitoring which size restocked because it would take much more resources and I don't have the fund to get a better server for the restock monitor :/

**Comment by manbearcoon** [comment]
Makes sense, thanks a lot for the quick feedback. How difficult would it be to add an audio notification in addition to the visual pop-up?

**Comment by val92130** [comment]
It would actually be pretty easy but I'm wondering if this won't be too anoying :/

**Comment by manbearcoon** [comment]
Ah true, maybe instead of sound it would make more sense to have Slack integration using Incoming Webhooks. That way your bot will notify not only desktop but also mobile. And the user can choose to use it or not.

I'm hearing it's not hard to implement slack, seeing lots of developers add it. What do you think?

Here's the details: https://api.slack.com/incoming-webhooks

**Comment by val92130** [comment]
That seems like a good idea! I've never used Slack but I might look into that :)

**Comment by Bleez103** [comment]
> For the moment I don't plan on monitoring which size restocked because it would take much more resources and I don't have the fund to get a better server for the restock monitor :/

@val92130 Could you release this as an standalone JS script and we can run it on our own servers?

**Comment by val92130** [comment]
@Bleez103 Yes I'm planning on releasing it as a standalone .exe application :) Would it work with your servers?

**Comment by Bleez103** [comment]
@val92130  I'm running an AWS server so it should and if your able to implement the slack support we'd be alerted to our channels once a restock occurs while it is being ran remotely.

**Comment by tehisaac** [comment]
Possible to have it function on mac as well perhaps with JAR?

**Comment by val92130** [comment]
@Bleez103 I'll add an issue regarding the Slack support and try to implement it!

@tehisaac It will be cross platform so it will work on Linux, Mac ans Windows :)

**Comment by Bleez103** [comment]
@val92130 .. It appears that the restock notifications have stopped worked with version 2.8.2

**Comment by val92130** [comment]
@Bleez103 Yeah Supreme started banning crawlers on their site but I think I found a way around, I'm fixing it right now, it should be working again in a few hours!

**Comment by Bleez103** [comment]
@val92130  Yeah i checked the code after I posted -- I should updated that part with my second edit. My bad.

That's great news that you found a workaround.

**Comment by val92130** [comment]
@Bleez103 it's okay don't worry! :)
I'm deploying the fix in a few minutes, could you check on your side if it's working by going to the 'product listing' tab please?

**Comment by val92130** [comment]
@Bleez103 should be working by now! Took longer than I thought

**Comment by Bleez103** [comment]
@val92130 Everything is up and running now. Thanks for the quick update on your side

**Comment by val92130** [comment]
@Bleez103 You're welcome! Thanks for notifying me of the issue :)
